packages/flutter/lib/src/widgets/raw_menu_anchor.dart

```diff
@@ -0,0 +1,1960 @@
+  }
+
+  FocusNode? get _firstFocus {
+    assert(_menuScopeNode != null, '_firstFocus requires a menu scope node.');
+    return _overlayTraversalPolicy?.findFirstFocus(
+      _menuScopeNode!,
+      ignoreCurrentFocus: true
+    );
+  }
+
+  FocusNode? get _lastFocus {
+    assert(_menuScopeNode != null, '_lastFocus requires a menu scope node.');
+    return _overlayTraversalPolicy?.findLastFocus(
+      _menuScopeNode!,
+      ignoreCurrentFocus: true
+    );
+  }
+
+  @override
+  bool get _isOpen => _overlayController.isShowing;
+
+  @override
+  void initState() {
+    super.initState();
+    // If the overlay is custom, then focus is handled externally.
+    if (!widget.hasExternalFocusScope) {
+      _menuScopeNode = FocusScopeNode(
+        debugLabel: kReleaseMode ? null : '${describeIdentity(this)} Sub Menu'
+      );
+      _menuFocusNode = FocusNode(
+        debugLabel: kReleaseMode ? null : '${describeIdentity(this)} Focus Node'
+      );
+    }
+  }
+
+  @override
+  void dispose() {
+    _menuScopeNode?.dispose();
+    _menuFocusNode?.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget _buildAnchor(BuildContext context) {
+    Widget child = Shortcuts(
+      includeSemantics: false,
+      shortcuts: _kMenuTraversalShortcuts,
+      child: TapRegion(
+        groupId: _root._menuController,
+        consumeOutsideTaps: _root._isOpen && widget.consumeOutsideTaps,
+        onTapOutside: _handleOutsideTap,
+        child: Builder(
+          key: _anchorKey,
+          builder: (BuildContext context) {
+            return widget.builder?.call(context, _menuController, widget.child)
+                    ?? widget.child
+                    ?? const SizedBox();
+          },
+        ),
+      ),
+    );
+
+    if (widget.hasExternalFocusScope) {
+      return OverlayPortal.targetsRootOverlay(
+        controller: _overlayController,
+        overlayChildBuilder: _buildOverlay,
+        child: child,
+      );
+    }
+
+    if (_isRootOverlay) {
+      child = Actions(
+        actions: _isOpen ? _rootOverlayAnchorActions : <Type, Action<Intent>>{},
+        child: child,
+      );
+    }
+
+    // Focus is only used to monitor focus changes, so it's not necessary to
+    // include semantics or allow focus to be requested.
+    return Focus(
+      focusNode: _menuFocusNode,
+      includeSemantics: false,
+      canRequestFocus: false,
+      onFocusChange: _handleFocusChange,
+      child: OverlayPortal.targetsRootOverlay(
+        controller: _overlayController,
+        overlayChildBuilder: _buildOverlay,
+        child: child,
+      ),
+    );
+  }
+
+  Widget _buildOverlay(BuildContext context) {
+    final BuildContext anchorContext = _anchorKey.currentContext!;
+    final RenderBox overlay = Overlay.of(anchorContext).context.findRenderObject()! as RenderBox;
+    final RenderBox anchor = anchorContext.findRenderObject()! as RenderBox;
+    final Rect anchorRect = anchor.localToGlobal(Offset.zero) & anchor.size;
```

Comment:
```suggestion
    final Rect anchorRect = anchor.localToGlobal(Offset.zero, ancestor: overlay) & anchor.size;
```

Just after posting my comment about the failing tests, I remembered seeing [this commit](https://github.com/flutter/flutter/pull/158255/commits/763bdeb255315edce1eb8447931c953f13d17773) and wondering why the ancestor parameter was removed?
Looks like it was required as there are no more test failures once this parameter is restored.